aten/src/ATen/native/quantized/cpu/qnnpack/src/tanh.c

```diff
@@ -108,7 +108,7 @@ enum pytorch_qnnp_status pytorch_qnnp_create_tanh_nc_q8(
     const float x =
         input_scale * (float)(i - (int32_t)(uint32_t)input_zero_point);
     /* Scale tanh(x) by 1 / output scale = 256.0 */
-    float scaled_tanh_x = 256.0f * tanh(x);
+    float scaled_tanh_x = 128.0f * tanh(x) + 128.0f;
```

Comment:
Do the qnnpack tests for tanh pass with this change (aten/src/ATen/native/quantized/cpu/qnnpack/test/tanh-operator-tester.h)? I'm guessing that needs to change as well.

Reply:
This is the new scale -- I just forgot to change the comment. The scale is `2.0 / 256 = 1.0/128`, and the zero_point is 128 -- I am assuming the UINT8. I will update the comment

Reply:
I was referring to the unit tests here -https://github.com/pytorch/pytorch/blob/cc319659e387634359c2f188ea3adefeb2bc2954/aten/src/ATen/native/quantized/cpu/qnnpack/test/tanh-operator-tester.h#L93. I believe the scale and zero point for output ref needs to be updated here as well.

Reply:
Oh my bad -- made the change, lemme test

Reply:
```console
$> ./scripts/build_local.sh -DBUILD_BINARY=ON -DBUILD_PYTHON=OFF -DUSE_OBSERVERS=OFF -DUSE_DISTRIBUTED=OFF
$> ./build/local/tanh-test
Running main() from gtest_main.cc
[==========] Running 18 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 18 tests from TANH_OP
[ RUN      ] TANH_OP.zero_batch
[       OK ] TANH_OP.zero_batch (0 ms)
[ RUN      ] TANH_OP.unit_batch
[       OK ] TANH_OP.unit_batch (1 ms)
[ RUN      ] TANH_OP.unit_batch_with_qmin
[       OK ] TANH_OP.unit_batch_with_qmin (0 ms)
[ RUN      ] TANH_OP.unit_batch_with_qmax
[       OK ] TANH_OP.unit_batch_with_qmax (0 ms)
[ RUN      ] TANH_OP.unit_batch_with_input_scale
[       OK ] TANH_OP.unit_batch_with_input_scale (1 ms)
[ RUN      ] TANH_OP.unit_batch_with_input_zero_point
[       OK ] TANH_OP.unit_batch_with_input_zero_point (1 ms)
[ RUN      ] TANH_OP.small_batch
[       OK ] TANH_OP.small_batch (0 ms)
[ RUN      ] TANH_OP.small_batch_with_input_stride
[       OK ] TANH_OP.small_batch_with_input_stride (1 ms)
[ RUN      ] TANH_OP.small_batch_with_output_stride
[       OK ] TANH_OP.small_batch_with_output_stride (0 ms)
[ RUN      ] TANH_OP.small_batch_with_qmin
[       OK ] TANH_OP.small_batch_with_qmin (0 ms)
[ RUN      ] TANH_OP.small_batch_with_qmax
[       OK ] TANH_OP.small_batch_with_qmax (0 ms)
[ RUN      ] TANH_OP.small_batch_with_input_scale
[       OK ] TANH_OP.small_batch_with_input_scale (1 ms)
[ RUN      ] TANH_OP.small_batch_with_input_zero_point
[       OK ] TANH_OP.small_batch_with_input_zero_point (1 ms)
[ RUN      ] TANH_OP.strided_batch
[       OK ] TANH_OP.strided_batch (1 ms)
[ RUN      ] TANH_OP.strided_batch_with_qmin
[       OK ] TANH_OP.strided_batch_with_qmin (0 ms)
[ RUN      ] TANH_OP.strided_batch_with_qmax
[       OK ] TANH_OP.strided_batch_with_qmax (0 ms)
[ RUN      ] TANH_OP.strided_batch_with_input_scale
[       OK ] TANH_OP.strided_batch_with_input_scale (2 ms)
[ RUN      ] TANH_OP.strided_batch_with_input_zero_point
[       OK ] TANH_OP.strided_batch_with_input_zero_point (1 ms)
[----------] 18 tests from TANH_OP (10 ms total)

[----------] Global test environment tear-down
[==========] 18 tests from 1 test case ran. (10 ms total)
[  PASSED  ] 18 tests.
```

Reply:
I am not 100% sure why it was passing before, honestly. I think the reason is that the reference is also using the same scale/zp as the implementation, so it has the same output range. The scale only affects the range, when the input is relatively large. I think what is happening is that given the scale/zp the reference produced the same results as the qtanh